src/main/java/hudson/plugins/mercurial/Cache.java

```diff
@@ -104,12 +108,18 @@ private synchronized ReentrantLock getLockForSlaveNode(String node) {
             if (master == null) { // Should not happen
                 throw new IOException("Cannot retrieve the Jenkins master node");
             }
-            final FilePath rootPath = master.getRootPath();
-            if (rootPath == null) {
-                throw new IOException("Cannot retrieve the root directory of the Jenkins master node");
+
+            FilePath masterCaches = null;
+            if (masterCacheRoot != null){
+                masterCaches = new FilePath(master.getChannel(), masterCacheRoot);
```

Comment:
How should I deal with the following 
`FilePath masterCache = masterCaches.child(hash);`
which is 9 lines lower ?